### PR TITLE
[#563] Components page accessibility issues 

### PR DESCRIPTION
**Pull request by pylapp**
# Issue

#563 

# Definition of Done

- [x] Low text and icon contrast: orange on grey - **won't fix yet, need design evolution**
- [x] Truncated component names in zoom 235% - fixed with 4e09e5c31acd896e25bb42f1303df4628cc5d6ad
- [x] Keyboard focus contrast <3:1 - **won't fix yet, failed to get focused state if keyboard used**
- [x] Non-interactive elements are focusable with switch control. **fiwed with 5e730a529b9811fcb3104c785e33b71ef58c13c8**
- [x]  No vocalization of title after displaying the page - fixed with 6acd22f556ac1cad77f2f612f40170ea9ede1242

### CODE REVIEW

File: OrangeDesignSystemDemo/OrangeDesignSystemDemo/Utils/UI/AccssibleNavigationTitleModifier.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// or see the "LICENSE" file for more details.
+// 
+// Authors: See CONTRIBUTORS.txt
+// Software description: A SwiftUI components library with code examples for Orange Design System 
+//
+
+import Foundation
+import SwiftUI
+
+struct AccssibleNavigationTitleModifier: ViewModifier {
```
Comment (ludovic35) [inline OrangeDesignSystemDemo/OrangeDesignSystemDemo/Utils/UI/AccssibleNavigationTitleModifier.swift:17]: add an helper:

extension de View {
   funk odsNavigatinoTitle(title: String) -> some View {
        self.modifier(AccssibleNavigationTitleModifier(title))
   }
}

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift [hunk 2]
```diff
@@ -92,7 +94,12 @@ public struct ODSAboutModule<TermsOfService>: View where TermsOfService: View {
                 .padding(.bottom, ODSSpacing.s)
         }
         .listStyle(.plain)
-        .navigationTitle(°°"modules.about.title")
+        .navigationTitle(title)
```
Comment (ludovic35) [inline OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift:97]: use  AccssibleNavigationTitleModifier

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift [hunk 3]
```diff
@@ -92,7 +94,12 @@ public struct ODSAboutModule<TermsOfService>: View where TermsOfService: View {
...
+        .onAppear {
```
Comment (ludovic35) [inline OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift:98]: We should aply it everywhere in (demo app, in lib) ?

File: OrangeDesignSystemDemo/OrangeDesignSystemDemo/Screens/Components/ComponentsList.swift [hunk 4]
```diff
@@ -55,15 +57,18 @@ struct ComponentsList: View {
     var body: some View {
         NavigationView {
             ScrollView {
-                LazyVGrid(columns: columns, spacing: ODSSpacing.none) {
-                    ForEach(components, id: \.id) { component in
-                        smallCard(for: component)
+                if sizeCategory.isAccessibilityCategory {
```
Comment (ludovic35) [inline OrangeDesignSystemDemo/OrangeDesignSystemDemo/Screens/Components/ComponentsList.swift:60]: Should we create a view builder ?
SmallCardAccessibilityAlignedCollection ?

**Review by ludovic35 — CHANGES_REQUESTED** [review]

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift (hunk 3 above)
Comment (pylapp) [reply OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift:98]: Yes, maybe. Each buggy case will be listed in dedicated card. So we may apply it in both lib and app.

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift [hunk 5]
```diff
@@ -92,7 +94,12 @@ public struct ODSAboutModule<TermsOfService>: View where TermsOfService: View {
                 .padding(.bottom, ODSSpacing.s)
         }
         .listStyle(.plain)
-        .navigationTitle(°°"modules.about.title")
+        .odsNavigationTitle(title)
+        .onAppear {
```
Comment (ludovic35) [inline OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift:98]: no need .onAppear if already done in modifier ?

**Review by ludovic35 — CHANGES_REQUESTED** [review]

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Utils/AccessibleNavigationTitleModifier.swift [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import SwiftUI
+
+/// `ViewModifier` which defines a navigation title for the calling `View` and also use `UIAccessibility` to notify for screen changed.
+/// The title can be a wording key which will be localized.
+struct AccssibleNavigationTitleModifier: ViewModifier {
+
+    private let title: String
+
+    init(title: String) {
+        self.title = title.localized()
```
Comment (ludovic35) [inline OrangeDesignSystem/Sources/OrangeDesignSystem/Utils/AccessibleNavigationTitleModifier.swift:24]: est -ce logique de faire le localized ici ?

**Review by ludovic35 — CHANGES_REQUESTED** [review]

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift [hunk 7]
```diff
@@ -92,7 +94,7 @@ public struct ODSAboutModule<TermsOfService>: View where TermsOfService: View {
                 .padding(.bottom, ODSSpacing.s)
         }
         .listStyle(.plain)
-        .navigationTitle(°°"modules.about.title")
+        .odsNavigationTitle(title)
```
Comment (ludovic35) [inline OrangeDesignSystem/Sources/OrangeDesignSystem/Modules/About/ODSAboutModule.swift:97]: .odsNavigationTitle("modules.about.title".🌐)

File: OrangeDesignSystem/Sources/OrangeDesignSystem/Utils/View+extension.swift [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+//
+
+import Foundation
+import SwiftUI
+
+extension View {
+
+    /// Adds a modifier to the current `View` so as to define a naivgation title using the current `title`
+    /// and also send a notification for accessibility layers for a change of screen when appeared.
+    /// - Parameter title: The wording or wording key for the navigation title
```
Comment (ludovic35) [inline OrangeDesignSystem/Sources/OrangeDesignSystem/Utils/View+extension.swift:21]: /// - Parameter title: The wording or wording key for the navigation title
 - Parameter title: The navigation title

**Review by ludovic35 — CHANGES_REQUESTED** [review]

**Review by ludovic35 — APPROVED** [review]

**Pull request merged**